test/position-managers/Execute.t.sol

```diff
@@ -119,7 +121,66 @@ contract ExecuteTest is Test, PosmTestSetup, LiquidityFuzzers {
     }
 
     // rebalance: burn and mint
-    function test_execute_rebalance() public {}
+    function test_execute_rebalance_perfect() public {
+        uint256 initialLiquidity = 100e18;
+
+        // mint a position on range [-300, 300]
+        BalanceDelta delta = mint(range, initialLiquidity, address(this), ZERO_BYTES);
+        uint256 tokenId = lpm.nextTokenId() - 1;
+
+        // we'll burn and mint a new position on [-60, 60]; calculate the liquidity units for the new range
+        LiquidityRange memory newRange = LiquidityRange({poolKey: range.poolKey, tickLower: -60, tickUpper: 60});
+        uint128 newLiquidity = LiquidityAmounts.getLiquidityForAmounts(
+            SQRT_PRICE_1_1,
+            TickMath.getSqrtPriceAtTick(newRange.tickLower),
+            TickMath.getSqrtPriceAtTick(newRange.tickUpper),
+            uint128(-delta.amount0()),
+            uint128(-delta.amount1())
+        );
+
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+
+        Planner.Plan memory planner = Planner.init();
+        planner = planner.add(Actions.DECREASE, abi.encode(tokenId, initialLiquidity, ZERO_BYTES));
+        planner = planner.add(Actions.BURN, abi.encode(tokenId));
+        planner = planner.add(Actions.MINT, abi.encode(newRange, newLiquidity, address(this), ZERO_BYTES));
+        bytes memory calls = planner.finalize(range.poolKey);
+
+        bytes[] memory data = lpm.modifyLiquidities(calls, _deadline);
+        int256 delta0 = abi.decode(data[data.length - 2], (int256));
+        int256 delta1 = abi.decode(data[data.length - 1], (int256));
+
+        uint256 balance0After = currency0.balanceOfSelf();
+        uint256 balance1After = currency1.balanceOfSelf();
+
+        // TODO: use clear so user does not pay 1 wei
+        assertApproxEqAbs(delta0, 0, 1 wei);
```

Comment:
is this currently 1 wei off? can you assert its exactly 1 wei off, that way this does not pass when we use clear